domain/application/service/unit.go

```diff
@@ -110,6 +111,10 @@ type UnitState interface {
 	// If the unit is dead, an error satisfying [applicationerrors.UnitIsDead]
 	// is returned.
 	SetUnitConstraints(ctx context.Context, inUnitUUID coreunit.UUID, cons constraints.Constraints) error
+
+	// SetRunningAgentBinaryVersion sets the running agent version for the unit.
+	// A [applicationerrors.UnitNotFound] error will be returned if the unit does not exist.
+	SetRunningAgentBinaryVersion(context.Context, string, agentbinary.Version) error
```

Comment:
The existing state APIs take `coreunit.UUID` as a parameter, not a string, so we need to be consistent.